sfml: 2.5.1 -> 3.0.0

GaetanLepage opened the pull request:
## Things done

- **sfml: move to by-name**
- **sfml: 2.5.1 -> 3.0.0, cleanup**

Changelog: https://github.com/SFML/SFML/blob/3.0.0/changelog.md

Fixes #383063, #384566 

cc @astsmtl

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
simplenes: use sfml_2

GaetanLepage committed:
antsimulator: cleanup, use sfml_2

GaetanLepage committed:
cutecapture: cleanup, use sfml_2

GaetanLepage committed:
attract-mode: use sfml_2

GaetanLepage committed:
extremetuxracer: use sfml_2

GaetanLepage committed:
liblapin: use sfml_2

GaetanLepage committed:
vbam: cleanup, use sfml_2

GaetanLepage committed:
qstopmotion: move to by-name

GaetanLepage committed:
qstopmotion: cleanup, use sfml_2

GaetanLepage committed:
guvcview: cleanup, use sfml_2

samuela commented on pkgs/development/python-modules/nocturne/default.nix:
```diff
@@ -1,27 +1,28 @@
   pname = "nocturne";
-  version = "unstable-2022-10-15";
-  format = "setuptools";
+  version = "0-unstable-2024-06-19";
```
why `0-unstable-*-*-*`? iiuc `unstable-*-*-*` is the recommended format

Sigmanificient replied:
https://github.com/NixOS/nixpkgs/blob/c1ee2620296430ac1e3ee72583ad0191463a9d60/pkgs/README.md?plain=1#L389

> *- It _must_ start with a digit. This is required > for backwards-compatibility with [how `nix-env` parses derivation names](https://nix.dev/manual/nix/latest/command-ref/nix-env#selectors).*
> 
>   *Example: `"0.3.1rc2"` or `"0-unstable-1970-01-01"`*
> 
> *- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest upstream version preceding that commit, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format.*

GaetanLepage committed:
sfml_2: init at 2.6.2 (ex sfml)

GaetanLepage committed:
sfml: 2.6.2 -> 3.0.0

Diff: https://github.com/SFML/SFML/compare/2.5.1...3.0.0

Changelog: https://github.com/SFML/SFML/blob/3.0.0/changelog.md

GaetanLepage committed:
python312Packages.nocturne: use sfml_2

GaetanLepage committed:
opendungeons: use sfml_2

GaetanLepage committed:
marble-marcher-ce: use sfml_2

GaetanLepage committed:
csfml: use sfml_2

GaetanLepage committed:
mars: use sfml_2

Sigmanificient commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 387764`

Logs: https://github.com/Sigmanificient/nixpkgs-review-gha/actions/runs/13859248506


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>shadered</li>
    <li>slade</li>
    <li>sladeUnstable</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 15 packages built:</summary>
  <ul>
    <li>antsimulator</li>
    <li>dolphin-emu</li>
    <li>dolphin-emu-primehack</li>
    <li>empty-epsilon</li>
    <li>libretro.dolphin</li>
    <li>lutris</li>
    <li>lutris-free</li>
    <li>openrw</li>
    <li>qstopmotion</li>
    <li>retroarch-free</li>
    <li>retroarch-full</li>
    <li>sfml</li>
    <li>sfml_2</li>
    <li>simplenes</li>
    <li>vbam</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>shadered</li>
    <li>slade</li>
    <li>sladeUnstable</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>retroarch-free</li>
    <li>retroarch-full</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>antsimulator</li>
    <li>dolphin-emu</li>
    <li>dolphin-emu-primehack</li>
    <li>empty-epsilon</li>
    <li>libretro.dolphin</li>
    <li>openrw</li>
    <li>qstopmotion</li>
    <li>sfml</li>
    <li>sfml_2</li>
    <li>simplenes</li>
    <li>vbam</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>dolphin-emu-primehack</li>
    <li>shadered</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>dolphin-emu</li>
    <li>openrw</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sfml</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>dolphin-emu-primehack</li>
    <li>shadered</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>dolphin-emu</li>
    <li>openrw</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sfml</li>
  </ul>
</details>

GaetanLepage committed:
dolphin-emu: mark as broken on darwin

SuperSandro2000 commented on pkgs/by-name/sf/sfml/package.nix:
I don't like that we are inheriting things from version 2. It could be in the future that we need to add workaround in the old version which are not relevant for version 3, and we don't want to degrade this package then.

pbsds replied:
I agree, it's better to copy it into here. Consider leaving a comment about their origin from sfml_2 if you think this is important.

SuperSandro2000 reviewed (commented):
otherwise lgtm

pbsds commented on pkgs/by-name/si/simplenes/package.nix:
```diff
@@ -13,24 +13,28 @@ stdenv.mkDerivation {
     repo = "SimpleNES";
-    rev = "4edb7117970c21a33b3bfe11a6606764fffc5173";
-    sha256 = "1nmwj431iwqzzcykxd4xinqmg0rm14mx7zsjyhcc5skz7pihz86g";
+    rev = "154a2fd4f2f2611a27197aa8d802bbcdfd1a0ea3";
```
Whats up with this rev change?

pbsds commented on pkgs/by-name/do/dolphin-emu/package.nix:
```diff
@@ -180,6 +180,10 @@ stdenv.mkDerivation (finalAttrs: {
     branch = "master";
     license = lib.licenses.gpl2Plus;
     platforms = lib.platforms.unix;
+    badPlatforms = [
+      # error: implicit instantiation of undefined template 'std::char_traits<unsigned int>'
+      lib.systems.inspect.patterns.isDarwin
+    ];
```
ack

pbsds commented:
https://github.com/NixOS/nixpkgs/pull/387764#discussion_r2008064202 is unaddressed, and i think it makes sense
